sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/sonar/TableWithoutHeaderCheck.java

```diff
@@ -46,12 +46,11 @@ private static boolean isHidden(TagNode node) {
   }
 
   private static boolean hasHeader(TagNode node) {
-    return node.getChildren().stream().filter(TableWithoutHeaderCheck::isTableRow).anyMatch(
-        row -> row.getChildren().stream().anyMatch(TableWithoutHeaderCheck::isTableHeader));
-  }
-
-  private static boolean isTableRow(TagNode node) {
-    return "TR".equalsIgnoreCase(node.getNodeName());
+    if (node.getChildren().stream().anyMatch(TableWithoutHeaderCheck::isTableHeader)) {
+      return true;
+    } else {
+      return node.getChildren().stream().filter(child -> !isTable(child)).anyMatch(TableWithoutHeaderCheck::hasHeader);
+    }
```

Comment:
You could use shortcircuit evaluation to avoid `if-else` block here:
```
return node.getChildren().stream().anyMatch(TableWithoutHeaderCheck::isTableHeader) ||
    node.getChildren().stream().filter(child -> !isTable(child)).anyMatch(TableWithoutHeaderCheck::hasHeader);
```